File: Platforms/Common/Reign.Video.XNA/States/SamplerState.cs
using Reign.Core;
using X = Microsoft.Xna.Framework.Graphics;

namespace Reign.Video.XNA
{
	public class SamplerState : Disposable, SamplerStateI
	{
		#region Properties
		private Video video;
		private SamplerStateDesc desc;
		#endregion

		#region Constructors
		public static SamplerState New(DisposableI parent, SamplerStateDescI desc)
		{
			return new SamplerState(parent, desc);
		}

		public SamplerState(DisposableI parent, SamplerStateDescI desc)
		: base(parent)
		{
			video = parent.FindParentOrSelfWithException<Video>();
			this.desc = (SamplerStateDesc)desc;
		}
		#endregion

		#region Methods
		public void Enable(int index)
		{
			var samplerState = new X.SamplerState()
			{
				Filter = desc.filter,
				MaxAnisotropy = 1,
				AddressU = desc.addressU,
				AddressV = desc.addressV,
				#if !SILVERLIGHT
				AddressW = desc.addressW
				#endif
			};

			video.Device.SamplerStates[index] = samplerState;
		}
		#endregion
	}
}

File: Platforms/Win32/Reign_Video_D3D9_Component/SamplerState.cpp
#include "pch.h"
#include "SamplerState.h"

namespace Reign_Video_D3D9_Component
{
	#pragma region Constructor
	SamplerStateCom::SamplerStateCom(VideoCom^ video, SamplerStateDescCom^ desc)
	{
		this->video = video;
		this->desc = desc;
	}
	#pragma endregion
	
	#pragma region Methods
	void SamplerStateCom::Enable(int index)
	{
		video->device->SetSamplerState(index, D3DSAMP_MINFILTER, desc->filter);
		video->device->SetSamplerState(index, D3DSAMP_MAGFILTER, desc->filter);
		video->device->SetSamplerState(index, D3DSAMP_MIPFILTER, desc->filter);

		video->device->SetSamplerState(index, D3DSAMP_ADDRESSU, desc->addressU);
		video->device->SetSamplerState(index, D3DSAMP_ADDRESSV, desc->addressV);
		video->device->SetSamplerState(index, D3DSAMP_ADDRESSW, desc->addressW);

		video->device->SetSamplerState(index, D3DSAMP_BORDERCOLOR, desc->borderColor);
	}
	#pragma endregion
}